src/Servers/Kestrel/test/InMemory.FunctionalTests/ChunkedResponseTests.cs

```diff
@@ -1187,6 +1191,126 @@ public async Task ChunkedWithBothPipeAndStreamWorks()
+            //AppContext.SetSwitch("Switch.Microsoft.AspNetCore.Server.Kestrel.DisableAutoChunk", false);
+
+            await using (var server = new TestServer(async httpContext =>
+            {
+                var response = httpContext.Response;
+
+                response.Headers.Add(HeaderNames.TransferEncoding, "chunked");
+                await response.WriteAsync("hello, world");
+                await response.WriteAsync("hello, world");
+                await response.WriteAsync("hello, world");
+
+            }, new TestServiceContext(LoggerFactory)))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.Send(
+                        "GET / HTTP/1.1",
+                        "Host:",
+                        "",
+                        "");
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "Transfer-Encoding: chunked",
+                        "",
+                        "c",
+                        "hello, world",
+                        "c",
+                        "hello, world",
+                        "c",
+                        "hello, world",
+                        "0",
+                        "",
+                        "");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task NoSwitchAutoChunkForChunedTransferEncodingWorks()
+        {
+            await using (var server = new TestServer(async httpContext =>
+            {
+                var response = httpContext.Response;
+
+                response.Headers.Add(HeaderNames.TransferEncoding, "chunked");
+                await response.WriteAsync("hello, world");
+                await response.WriteAsync("hello, world");
+                await response.WriteAsync("hello, world");
+
+            }, new TestServiceContext(LoggerFactory)))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.Send(
+                        "GET / HTTP/1.1",
+                        "Host:",
+                        "",
+                        "");
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "Transfer-Encoding: chunked",
+                        "",
+                        "c",
+                        "hello, world",
+                        "c",
+                        "hello, world",
+                        "c",
+                        "hello, world",
+                        "0",
+                        "",
+                        "");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task Http2Chun()
```

Comment:
Remove this test. HTTP/2 requests can't be written like this, the message encoding is completely different.